### DISCUSSION TITLE
Concomitant installation of old and new MOPACs

### DISCUSSION THREAD

**Discussion by icamps** [openmopac/mopac#91] [Q&A] [answered]
Hello,

Is it possible to have both versions of MOPAC installed/working at the same time?

I mean, I would like to run some test with both versions just running from the command prompt something like `old_MOPAC input.mop` and `new_MOPAC input.mop`, but I am not sure what to do with the symlinks to the library files.

Regards,

Camps

**Comment by godotalgorithm** [comment, marked as answer]
In principle, this should work just by making all the desired versions available in the path and giving different names to the different executables. On Mac/Linux, I have connected the executables and shared libraries with a relative RPATH, so the executable should prioritize the libmopac version that is located in `../lib` relative to the executable. On Windows, it should simply prioritize the DLL in the local directory before remote ones as far as I understand. If there is deviation from this intended/expected behavior, then please report it and I'll examine it further.

This would be more complicated if you want all of the different library and executable versions in the same directories. In that case, you'd have to give different names to the different library versions, and you'd need to adjust the library names that the executable is looking for using OS-appropriate tools (`patchelf`, `install_name_tool`).

**Reply by icamps** [reply]
I'm not aware about using RPATH, thanks for the suggestion.

**Comment by godotalgorithm** [comment]
There is an extra caveat that has come up in recent support emails. On Linux, the path to the libraries are set by RUNPATH rather than RPATH, because that is what CMake supports. RUNPATH can be overridden by the LD_LIBRARY_PATH environment variable, so it is important that none of the MOPAC library directories (or old versions of `libiomp5.so`) appear in LD_LIBRARY_PATH or they might override the other MOPAC versions.
